content/developer/reference/frontend/services.rst

```diff
@@ -347,6 +349,144 @@ Here, it is called in webclient.js to make it visible everywhere for the example
+
+.. note::
+   Hotkeys are not case sensitive but they must comply to these rules:
+
+   - each part is separated by the "+" character
+   - all parts are whitelisted
+   - "control", "shift" and "alt" modifiers:
+
+     - are optional
+     - can be combined
+     - must come first
+
+   Whitelist:
+
+   - Alphanumerics: a-z 0-9
+   - Navigational:
+
+     - "arrowleft", "arrowright", "arrowup", "arrowdown"
+     - "home", "end", "pageup", "pagedown"
+     - "tab", "enter", "escape", "backspace", "delete"
+
+   Some examples:
+
+   - "shift+tab"
+   - "tab" (modifiers are optional)
+   - "alt+control+d", "Control+Alt+D", "CONTROL+alt+d" are equivalent
+
+.. admonition:: Great To Know
+
+   The Hotkey Service does not support numerics keys coming from the NumPad
+   but only from the DigitRow above the letters.
+
+   This is due to an incompatible input method of ASCII characters
+   on Windows operating systems: `ALT+(numerical code from numpad)`.
+
+   See `this Microsoft documentation <https://support.microsoft.com/en-us/office/insert-ascii-or-unicode-latin-based-symbols-and-characters-d13f58d3-7bcb-44a7-a4d5-972ee12e50e0#bm1>`_.
+
+.. admonition:: Marvellous To Know !
+
+   When you define an hotkey somewhere you will
+   never have to worry about MacOS operating systems !
+
+   The Hotkey Service translates automatically hotkeys in order to
+   enhance the MacOS users experience as they are used to:
+
+   - using `Command` instead of `Control`
+   - using `Control` instead of `Alt`
+
+API
+~~~
+
+.. js:function:: add(hotkey, callback[, options])
+  
+    :param string hotkey: the hotkey we would like to get notified for
+    :param function callback: the hotkey we would like to get notified for
+    :param object options: the options of the notification
+    :returns: a function to call to clear the callback from the service
+
+    Adds a callback to be executed when the corresponding hotkey is pressed on the keyboard.
+
+    .. tip:: 
+       If you use this `add` function directly you HAVE to make sure you clear the callback at the
+       proper moment.
+       It is recommended you use the :ref:`useHotkey hook <frontend/hooks/usehotkey>` instead.
+
+    The options are defined by:
+
+    .. list-table::
+      :widths: 20 15 10 55
+      :header-rows: 1
+
+      * - Name
+        - Type
+        - Default value
+        - Description
+      * - `allowRepeat`
+        - boolean
+        - false
+        - The callback will continue to get called as long as the hotkey is being pressed.
+          Typical usage: arrows and other navigational keys.
+      * - `bypassEditableProtection`
+        - boolean
+        - false
+        - If true the hotkey service will call this registration
+          even if an editable element is focused
+      * - `global`
+        - boolean
+        - false
+        - A global hotkey callback will always get called no matter the UI active element.
+
+.. js:function:: registerIframe(iframe)
```

Comment:
no way to unregister a frame? isn't that a memory leak?

Reply:
When the iframe is removed from the DOM the listeners attached to its contentWindow are removed too